### ISSUE TITLE
[a11y] TextInput character count should be read aloud

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#12429]
Continuation of https://github.com/carbon-design-system/carbon/issues/12071 - the implementation for TextArea in https://github.com/carbon-design-system/carbon/pull/12411 needs to be abstracted and used in TextInput.

https://github.com/carbon-design-system/carbon/issues/12429#tasklist-block-29110d0a-7aa9-4a90-ba1d-d636e6630e9c

**Comment by Shankar-CodeJunkie** [comment]
@tay1orjones :  I would like to work on this issue . Thanks ! .

**Comment by tw15egan** [comment]
@Shankar-CodeJunkie all yours!

**Comment by Shankar-CodeJunkie** [comment]
@tay1orjones :  I created a pull request #12729 for your review and please let me know if there is any changes needed

**Comment by tay1orjones** [comment]
Closed via https://github.com/carbon-design-system/carbon/pull/12729, looks like the auto-link was missed due the space between the hash and the issue number `Closes # 12729`
